lib/Doctrine/DBAL/Connection.php

```diff
@@ -584,9 +584,13 @@ public function delete($tableExpression, array $identifier, array $types = array
         $paramValues = array();
 
         foreach ($identifier as $columnName => $value) {
-            $columnList[] = $columnName;
-            $criteria[] = $columnName . ' = ?';
-            $paramValues[] = $value;
+            if (is_null($value)) {
```

Comment:
I don't see how without passing columnList, criteria and paramValues as references to the helper method. (Which is probably why they're duplicated in the first place) Is that ok?